src/Search/Search.tsx

```diff
@@ -168,25 +186,67 @@ class SearchContainer extends React.Component<{}, ISearchState> {
   }
   private onResetForm() {
     this.setState({ query: [] });
-    this.updateQueryURL([]);
+    this.updateQueryURL([], this.state.searchBar);
   }
 
   private onFilterChange(newFilters: ISearchParameter[]) {
     this.setState({
       query: newFilters,
     });
-    this.updateQueryURL(newFilters);
+    this.updateQueryURL(newFilters, this.state.searchBar);
     this.triggerSearch();
   }
 
-  private updateQueryURL(newFilters: ISearchParameter[]) {
-    const newSearch = `?q=${encodeURIComponent(JSON.stringify(newFilters))}`;
+  private onSearchBarChanged(e: any) {
+    const searchBar = e.target.value;
+    this.setState({ searchBar });
+    this.updateQueryURL(this.state.query, searchBar);
+  }
+
+  private updateQueryURL(filters: ISearchParameter[], searchBar: string) {
+    const newSearch = `?q=${encodeURIComponent(
+      JSON.stringify(filters)
+    )}&searchBar=${encodeURIComponent(searchBar)}`;
     window.history.replaceState(
       null,
       '',
       window.location.href.split('?')[0] + newSearch
     );
   }
+
+  private filter(
+    results: Array<{
+      selected: boolean;
+      hacker: IHacker;
+    }>,
+    search: string
+  ): Array<{
+    selected: boolean;
+    hacker: IHacker;
+  }> {
+    return results.filter(({ hacker }) => {
+      const { accountId } = hacker;
+      const foundAcct =
+        typeof accountId !== 'string'
+          ? accountId.firstName.includes(search) ||
```

Comment:
I would make a temporary `name`(first last) variable to check through so that the search works if you're typing out someone's full name.

Reply:
Word that's fair!